src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java

```diff
@@ -596,6 +633,87 @@ public OptionData setRequiredRange(double minValue, double maxValue)
+        this.minLength = minLength;
+        return this;
+    }
+
+    /**
+     * Configure the maximum length for strings which can be provided for this option.
+     *
+     * @param  maxLength
+     *         The maximum length for strings which can be provided for this option.
+     * @throws IllegalArgumentException
+     *         If any of the following checks fail
+     *         <ul>
+     *             <li>{@link OptionType type of this option} is {@link OptionType#STRING STRING}</li>
+     *             <li>{@code maxLength} is greater than or equal to {@value MAX_STRING_OPTION_LENGTH_MINIMUM} and
+     *             lower than or equal to {@value MAX_STRING_OPTION_LENGTH}</li>
+     *         </ul>
+     *
+     * @return The OptionData instance, for chaining
+     */
+    @Nonnull
+    public OptionData setMaxLength(int maxLength)
+    {
+        if (type != OptionType.STRING)
+            throw new IllegalArgumentException("Can only set max length for options of type STRING");
+        Checks.check(maxLength >= MAX_STRING_OPTION_LENGTH_MINIMUM && maxLength <= MAX_STRING_OPTION_LENGTH,
+                "Max length must be greater than or equal to %d and lower than or equal to %d", MAX_STRING_OPTION_LENGTH_MINIMUM, MAX_STRING_OPTION_LENGTH);
+        this.maxLength = maxLength;
+        return this;
+    }
+
+    /**
+     * Configure the minimum and maximum length for strings which can be provided for this option.
+     *
+     * @param  minLength
+     *         The minimum length for strings which can be provided for this option.
+     * @param  maxLength
+     *         The maximum length for strings which can be provided for this option.
+     * @throws IllegalArgumentException
+     *         If any of the following checks fail
+     *         <ul>
+     *             <li>{@link OptionType type of this option} is {@link OptionType#STRING STRING}</li>
+     *             <li>{@code minLength} is greater than or equal to {@code 0}</li>
+     *             <li>{@code maxLength} is greater than or equal to {@value MAX_STRING_OPTION_LENGTH_MINIMUM} and
+     *             lower than or equal to {@value MAX_STRING_OPTION_LENGTH}</li>
+     *         </ul>
+     *
+     * @return The OptionData instance, for chaining
+     */
+    @Nonnull
+    public OptionData setRequiredLength(int minLength, int maxLength)
+    {
+        if (type != OptionType.STRING)
+            throw new IllegalArgumentException("Can only set min and max length for options of type STRING");
+        Checks.check(minLength >= 0, "Min length must be greater than or equal to 0");
+        Checks.check(maxLength >= MAX_STRING_OPTION_LENGTH_MINIMUM && maxLength <= MAX_STRING_OPTION_LENGTH,
+                "Max length must be greater than or equal to %d and lower than or equal to %d", MAX_STRING_OPTION_LENGTH_MINIMUM, MAX_STRING_OPTION_LENGTH);
+        this.minLength = minLength;
+        this.maxLength = maxLength;
+        return this;
+    }
```

Comment:
This should probably also check if `minLength` is greater than `maxLength`

Reply:
this isn't done in the range methods either, so I'd much rather leave this as is and maybe make a followup PR if you really wanted to and change it for both range and length methods.

Reply:
Currently this is done during the serialization for min and max value.

Reply:
Well, that's interesting. I definitely remember reading code that was doing the min/max comparison at serialization time because I remember deleting a comment i left on the min/max PR about adding that check in on the setX methods themselves but realized we were doing it just before serialization.

We definitely need to add these in. Them not being there for the others doesn't mean we shouldn't add it for these. I see no reason why we wouldn't add them tbh.

Reply:
where do I put it? putting the check into the `setXLength` methods doesn't really make sense because you can't know the order of these calls, and I doubt it's good practice to throw in `toData()`.

Reply:
That's how i did it in `TextInput`:
https://github.com/DV8FromTheWorld/JDA/blob/6e8c030fbe000c16224f658b1e37dffabe5a9383/src/main/java/net/dv8tion/jda/api/interactions/components/text/TextInput.java#L341-L350

In the cases where users use `setMaxLength()` and `setMinLength()`, it's checked in the `build()` method.